.github/actions/cache-rust/action.yaml

```diff
@@ -78,6 +78,23 @@ runs:
       run: |-
         error_code="0"
 
+        for value in \
+          "${cache_cargo}" \
+          "${cache_rust_stable}" \
+          "${cache_rust_nightly}" \
+          "${fail_on_cache_miss}"
+        do
+          case "${value}" in
+          ("false"|"true") ;;
+          (*)
+            "echo" "Expected boolean value for parameter, but got: \"${value}\"\
+        !"
+
+            error_code="1"
+            ;;
+          esac
+        done
+
```

Comment:
Do you have access to the source tree at this point? If yes, let's define a function that checks a single boolean value and call it as many times as we need here.